homework-2/services/userService.ts

```diff
@@ -1,83 +1,57 @@
-        age: 3,
-        isDeleted: false
-    },
-    {
-        id: '4b4ba4ab-0f8c-4669-9aef-0bcf220dad8e',
-        login: 'user4',
-        password: 'user4',
-        age: 4,
-        isDeleted: false
-    }
-];
+import { IUserAttrs } from "../interfaces/IUserAttrs";
+import { IUserModel } from "../models/userModel";
+import { db } from "../models";
 
-class UserService {
-    private usersList: IUser[];
 
-    constructor() {
-        this.usersList = initialUsersList;
-    }
-
-    getAllUsers(): IUser[] {
-        return this.usersList;
+class UserService {
+    getAllUsers(): Promise<IUserAttrs[]> {
+        return db.User.findAll()
     }
 
-    getUserById(id: string): IUser | null {
-        return this.usersList.find(user => user.id === id) || null;
+    getUserById(id: string): Promise<IUserModel | null> {
+        return db.User.findByPk(id);
     }
 
-    addUser(user: IUser): string {
-        user.id = user.id || uuidv4();
-        this.usersList.push(user);
+    async addUser(user: IUserAttrs): Promise<number> {
+        const existsByLogin = await db.User.findOne({ where: { login: user.login } })
+        if (!existsByLogin) {
+            const createdUser = await db.User.create(user);
 
-        return user.id;
+            return createdUser.id;
+        } else {
+            throw new Error('Validation error user with this login already exists!')
+        }
     }
 
-    updateUser(user: IUser): void {
-        const targetIndex = this.usersList.findIndex(user => user.id === user.id);
-        this.usersList[targetIndex] = user;
+    updateUser(user: IUserAttrs): void {
+        db.User.update(user, { where: { id: user.id }})
     }
 
-    isUserExists(userId: string): boolean {
-        return !!this.usersList.find(user => user.id === userId);
+    async isUserExists(userId: string): Promise<boolean> {
+        const user = await this.getUserById(userId);
+
+        return !!user;
     }
 
-    markUserAsDeleted(userId: string): void {
-        const user = this.getUserById(userId);
-        if (user) {
-            user.isDeleted = true;
+    async markUserAsDeleted(userId: string): Promise<void> {
+        const userToDelete = await this.getUserById(userId);
+
+        if (userToDelete && !userToDelete.isDeleted) {
+            userToDelete.isDeleted = true;
+            userToDelete.save();
         }
     }
 
-    getAutoSuggestUsers(loginSubstring: string, limit: number): IUser[] {
-        return this.usersList
-            .sort(this.compareByLogin)
+    async getAutoSuggestUsers(loginSubstring: string, limit: number): Promise<IUserAttrs[]> {
+        const users = await this.getAllUsers();
+
+        return users
+            .sort(UserService.compareByLogin)
             .filter(user => user.login.includes(loginSubstring))
             .slice(0, limit);
     }
 
-    private compareByLogin(u1: IUser, u2: IUser): number {
+    private static compareByLogin(u1: IUserAttrs, u2: IUserAttrs): number {
```

Comment:
Please use this string method
https://www.w3schools.com/jsref/jsref_localecompare.asp

Reply:
Done, thanks